src/main/java/com/redhat/cloud/policies/app/rest/PolicyCrudService.java

```diff
@@ -808,10 +809,10 @@ public Response getTriggerHistoryForPolicy(@PathParam("id") UUID policyId) {
                 Pager pager = PagingUtils.extractPager(uriInfo);
                 builder = buildHistoryResponse(policyId, pager);
             } catch (IllegalArgumentException iae) {
-                tracer.activeSpan().setTag(ERROR_STRING, true);
+                Span.current().setAttribute(ERROR_STRING, true);
```

Comment:
This code should be improved. 
Actually you sent  `span.setStatus(ERROR);`It's a new method in OpenTelemetry.
You should also record the exception.
Instead of calling  Span.current() multiple times, you should probably get the current span and use the try with resources to open a scope.